simulator/kernel/mars/t/unit_test.cpp

```diff
@@ -236,6 +236,27 @@ TEST_CASE( "MARS: open non existing file")
     CHECK( sys.sim->read_cpu_register( v0) == all_ones<uint64>());
 }
 
+static void write_buff_to_file
+(System* sys, uint64 descr, uint64 buff_ptr, uint64 chars_to_write)
+{
+  sys.sim->write_cpu_register( v0, 15u); //write to file
+  sys.sim->write_cpu_register( a0, descr);
+  sys.sim->write_cpu_register( a1, buff_ptr);
+  sys.sim->write_cpu_register( a2, chars_to_write);
+  CHECK( sys.mars_kernel->execute() == Trap::NO_TRAP);
+}
+
+static uint64 open_file(System* sys, uint64 filename_ptr, uint64 flags)
+{
+  sys.sim->write_cpu_register( v0, 13u); // open file
+  sys.sim->write_cpu_register( a0, filename_ptr);
+  sys.sim->write_cpu_register( a1, flags);
+  CHECK( sys.mars_kernel->execute() == Trap::NO_TRAP);
+  uint64 descriptor = sys.sim->read_cpu_register( v0);
+  CHECK( descriptor >= 3); // May be print line from it was called?
+  return descriptor;
```

Comment:
Can we have a separate test for file opening instead of making checks inside the helper functions?

```c++
TEST( "MARS: open file")
{
    auto trap = open_file( /* ... */);
    CHECK( trap == Trap::NO_TRAP);
    CHECK( get_descriptor() >= 3);
    // Should close as well?
}

TEST( "MARS: open file and write")
{
    open_file( /* ... */);
    auto descriptor = get_descriptor();
    auto trap = write_file( /* ... */ );
    CHECK( get_descriptor() >= 3);
    // Should close as well?
}

TEST_CASE( "MARS: open, write, read and close a file")
{
// ... 
```